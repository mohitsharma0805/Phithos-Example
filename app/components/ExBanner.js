import React from 'react';
import { View, Text, StyleSheet, Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import config from '../config';

const ExBanner = () => {
  const getDeviceType = () => {
    let deviceType = DeviceInfo.getModel();
    if (DeviceInfo.isEmulatorSync) {
      if (Platform.OS === 'ios') {
        deviceType = 'Simutator';
      } else {
        deviceType = 'Emulator';
      }
    }
    return deviceType;
  };

  return (
    <View style={styles.contentView}>
      <Text style={styles.text}>{getDeviceType()}</Text>
    </View>
  );
};

export default ExBanner;

const styles = StyleSheet.create({
  contentView: {
    justifyContent: 'center',
    height: 40,
    width: '100%',
    backgroundColor: config.colors.black,
  },
  text: { marginLeft: 20, color: config.colors.white, fontSize: 17 },
});
