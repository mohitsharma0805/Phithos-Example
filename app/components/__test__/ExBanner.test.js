import React from 'react';
import { shallow } from 'enzyme';
import { View, Text } from 'react-native';
import ExBanner from '../ExBanner';

describe('ExBanner', () => {
  describe('Rendering', () => {
    const wrapper = shallow(<ExBanner />);
    it('renders the container view', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(View).length).toEqual(1);
      expect(wrapper.find(Text).length).toEqual(1);
    });
  });
});
