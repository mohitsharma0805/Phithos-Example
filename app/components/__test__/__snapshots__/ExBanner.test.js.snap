// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExBanner Rendering renders the container view 1`] = `ShallowWrapper {}`;
